modules/navigation/src/main/java/com/sygic/maps/module/navigation/viewmodel/NavigationFragmentViewModel.kt

```diff
@@ -0,0 +1,151 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package com.sygic.maps.module.navigation.viewmodel
+
+import android.app.Application
+import android.os.Bundle
+import androidx.annotation.RestrictTo
+import androidx.lifecycle.DefaultLifecycleObserver
+import androidx.lifecycle.LifecycleOwner
+import androidx.lifecycle.MutableLiveData
+import com.sygic.maps.module.common.theme.ThemeManager
+import com.sygic.maps.module.common.viewmodel.ThemeSupportedViewModel
+import com.sygic.maps.module.navigation.KEY_PREVIEW_MODE
+import com.sygic.maps.module.navigation.component.PREVIEW_MODE_DEFAULT_VALUE
+import com.sygic.maps.tools.annotations.Assisted
+import com.sygic.maps.tools.annotations.AutoFactory
+import com.sygic.maps.uikit.viewmodels.common.location.LocationManager
+import com.sygic.maps.uikit.viewmodels.common.navigation.RouteDemonstrationManager
+import com.sygic.maps.uikit.viewmodels.common.permission.PermissionsManager
+import com.sygic.maps.uikit.viewmodels.common.sdk.model.ExtendedCameraModel
+import com.sygic.maps.uikit.viewmodels.common.sdk.model.ExtendedMapDataModel
+import com.sygic.maps.uikit.viewmodels.common.utils.requestLocationAccess
+import com.sygic.maps.uikit.views.common.extensions.getBoolean
+import com.sygic.sdk.map.Camera
+import com.sygic.sdk.map.MapAnimation
+import com.sygic.sdk.map.MapCenter
+import com.sygic.sdk.map.MapCenterSettings
+import com.sygic.sdk.map.`object`.MapRoute
+import com.sygic.sdk.navigation.NavigationManager
+import com.sygic.sdk.route.RouteInfo
+
+private const val DEFAULT_NAVIGATION_TILT = 60f
+private val DEFAULT_NAVIGATION_MAP_CENTER = MapCenter(0.5f, 0.3f)
+private val DEFAULT_NAVIGATION_MAP_CENTER_SETTING = MapCenterSettings(
+    DEFAULT_NAVIGATION_MAP_CENTER,
+    DEFAULT_NAVIGATION_MAP_CENTER,
+    MapAnimation.NONE,
+    MapAnimation.NONE
+)
+
+@AutoFactory
+@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
+class NavigationFragmentViewModel internal constructor(
+    app: Application,
+    @Assisted arguments: Bundle?,
+    themeManager: ThemeManager,
+    private val cameraModel: ExtendedCameraModel,
+    private val mapDataModel: ExtendedMapDataModel,
+    private val locationManager: LocationManager,
+    private val permissionsManager: PermissionsManager,
+    private val navigationManager: NavigationManager,
+    private val routeDemonstrationManager: RouteDemonstrationManager
+) : ThemeSupportedViewModel(app, themeManager), DefaultLifecycleObserver, NavigationManager.OnRouteChangedListener {
+
+    val previewMode: MutableLiveData<Boolean> = object : MutableLiveData<Boolean>() {
+        override fun setValue(value: Boolean) {
+            super.setValue(value)
+            routeInfo.value?.let { processRouteInfo(it) }
```

Comment:
nebolo by spravnejsie na to observovat? to routeInfo s nastavovanim dat nema moc spolocne

Reply:
v tom pod tym vyuzivas tu value, ale tu nie